cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -298,10 +304,9 @@ public StreamConfig call() throws Exception {
   }
 
   private void assertNamespaceHomeExists(Id.Namespace namespaceId) throws IOException {
-    Location namespaceHomeLocation = Locations.getParent(getStreamsHomeLocation(namespaceId));
-    Preconditions.checkArgument(namespaceHomeLocation.exists(),
-                                String.format("Home directory %s for namespace %s not found",
-                                              namespaceHomeLocation.toURI().getPath(), namespaceId));
+    Location namespaceHomeLocation = Locations.getParent(getNamespaceLocation(namespaceId));
+    Preconditions.checkArgument(namespaceHomeLocation == null || namespaceHomeLocation.exists(),
```

Comment:
good catch. I was thinking about it as a `if`. Precondition should flip the condition.